src/hooks/useGetBookFromUrl.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react'
+import { GoogleBook } from '../components/API/GoogleBook'
+import { BooksApiResponse, Items, VolumeInfo } from '../types/GoogleBooks'
+
+export const useGetBookFromUrl = (bookId: string) => {
```

Comment:
a lot of things are duplicated from the useSearchBooks hook

create a reusable hook and use it in both useGetBookFromUrl and useSearchBooks  hooks instead of duplicating code